Tests/interop/net/field/test_instance_fields.py

```diff
@@ -179,35 +181,38 @@ def f22(): v.InstanceSimpleClassField = SimpleClass(540)
         def f23(): v.InstanceSimpleGenericClassField = SimpleGenericClass[str]("string".upper())
         def f24(): v.InstanceSimpleInterfaceField = ClassImplementSimpleInterface(78)
 
-        funcs = [ eval("f%s" % i) for i in range(1, 25) ]
-        if clr.GetClrType(t).IsValueType:
-            for f in funcs:
-                self.assertWarns(RuntimeWarning, 'may result in updating a copy', f)
-        else: 
-            for f in funcs: f()
-            self._test_verify(o)
-            
+        for i in range(1, 25):
+            f = eval("f%s" % i)
+            if clr.GetClrType(t).IsValueType:
+                self.assertWarnsRegex(RuntimeWarning, 'may result in updating a copy', f)
+            else:
+                f()
+
+        if not clr.GetClrType(t).IsValueType:
+            self._test_verify(v)
+
+
             # set values which need conversion.
-            v.InstanceInt32Field = 100
-            self.assertEqual(o.InstanceInt32Field, 100)
+            v.InstanceInt32Field = long(100)
+            self.assertEqual(v.InstanceInt32Field, 100)
```

Comment:
I'm not sure about the original intent, but I think this should remain `o` to check that changes on the value are reflected by the boxed object?
```suggestion
            self.assertEqual(o.InstanceInt32Field, 100)
```